index.js

```diff
@@ -0,0 +1,130 @@
+
+const URL = "https://trektravel.herokuapp.com/trips"
+
+
+const loadTrips = () => {
+
+  tripList = $('#trip-list');
+  tripList.empty();
+
+
+  axios.get(URL)
+    .then((response) => {
+      response.data.forEach((trip) => {
+        tripList.append(`<li id="${trip.id}"><a href="#">${trip.name}</a></li>`);
+        $("#" + trip.id).click( () => {
+          loadDetails(trip.id);
+          createForm(trip.id, trip.name) })
+
+
+        })
+        tripList.prepend("<h2>All Trips</h2>")
+        $(".current-trips").css("border-style", "solid");
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+
+};
+
+
+const loadDetails = (id) => {
+  const tripDetails = $('#detail-list');
+  tripDetails.empty()
+
+  axios.get(URL+"/"+ id)
+  .then((response) => {
+      let titles = ["name", "continent", "category", "weeks", "cost", "about"]
+      titles.forEach((title) => {
+        tripDetails.append(`<li><b>${title}:</b> ${response["data"][title]}</li>`);
+      })
+      tripDetails.prepend(`<h2>Trip Details</h2>`)
+      $(".details").css("border-style", "solid");
+
+  })
+  .catch((error) => {
+    console.log(error);
+  });
+  
+};
+
+
+const createForm = (id, name) => {
+  const form = $('#form');
+  form.empty()
+  const buildFormHtml = `<h2>Reserve a Spot on ${name} </h2>
+  <div >Name: <input type="text" name="name"></div>
+  <div>Email: <input type="text" name="email"></div>
+  <input type="submit" name="reserve" value="Reserve" />`
+
+  form.html(buildFormHtml)
+  $(".reservation-form").css("border-style", "solid");
+
+  $('#form').submit( (event) => {
+    event.preventDefault();
+    createReservation(id, name)
+})
+}
+
+
+const createReservation = (id, name) => {
+
+  const reservationData = readFormData();
+
+  reportStatus('Sending reservation data...');
+
+  axios.post(URL+"/"+ id + "/reservations", reservationData)
+    .then((response) => {
+      console.log(response);
+      reportStatus(`Successfully added a reservation for ${name} !`);
+      clearForm();
+    })
+    .catch((error) => {
+      console.log(error.response);
+      reportStatus(`Encountered an error: ${error.message}`);
+    });
```

Comment:
The error sent back from the API has a bunch of data about why it didn't work that you're not reporting here.